src/main/java/xogame/app/RoundBuffer.java

```diff
@@ -0,0 +1,25 @@
+package xogame.app;
+
+public class RoundBuffer {
+    private Player[] players;
+
+    public RoundBuffer() {
+        players = new Player[2];
+    }
+
+    public void addPlayer(Player player) throws UnsupportedOperationException{
+        if (players[0] == null) {
+            players[0] = player;
+        } else if (players[1] == null) {
+            players[1] = player;
+        } else throw new UnsupportedOperationException("You cannot add more than two players.");
+    }
+
+    public Player takePlayer() {
+        Player temporaryPlayer = players[1];
+        Player resultPlayer = players[0];
+        players[0] = temporaryPlayer;
+        players[1] = resultPlayer;
+        return resultPlayer;
+    }
```

Comment:
Tworzone są nowe obiekty (referencje), więc  pewnie minimalnie wpływa na wydajność, ale działa i to się liczy :)